app/views/admin/pages/index.html.erb

```diff
@@ -36,13 +33,63 @@
 		<td>
 			<%= page.name %>
 		</td>
-		<td>
+		<td class="actions">
       <%= link_to 'View', url_for( "/pages/#{section.slug}/#{page.slug}" ) %>
-      <%= link_to 'Edit', url_for( controller: '/admin/pages', action: 'edit', id: page.id ) %>
+      <%= link_to 'Edit', url_for( controller: '/admin/pages', action: 'edit', id: page ) %>
 		</td>
 	</tr>
-	<% end %>
-	<% end %>
+
+  <% indent = indent + 2 %>
+  <% section.sections.each do |subsection| %>
```

Comment:
The table row/cell generation code looks like a good candidate for DRYing up with partial template(s).

Reply:
Yeah, particularly as it's a recursive data structure ("It's turtles all the way down!")

Reply:
Boom! https://github.com/denny/ShinyCMS-ruby/pull/35/commits/5aaf4715161b8aced8c1d2f93956543c2b8560c8